### ISSUE TITLE
Passwords error when starting NC Talk Call

### ISSUE THREAD

**Issue by simonspa** [marius-wieschollek/passwords#258]
Somehow there is an interference between Nextcloud Talk, Passwords and the Impersonate functionality.

Trying to test the calling functionality I impersonate a user and try to start a call by opening a private conversation with another user and click on "start call". Firefox then presents me with the "Do you really want to leave this page?" and upon clicking "yes" I end up on the NC error page pointing me to the `reqId` in the log, see below.

This error comes form Passwords somehow, which is why I post it here.

### System Information
- Nextcloud Version: 18.0.3
- PHP Version: 7.3
- Database and Version: MySQL 10.3
- Passwords Version: 2020.3.1
- Browser and Version: Firefox 74.0
- Client OS and Version: Ubuntu 19.10
- Server OS and Version: Debian 10

### Steps to reproduce
1. Have NC installed with Passwords and Talk
2. Impersonate another user
3. Try to start a call to another user
4. Observe error by NC

### Expected result
Passwords should not interfere with Talk.

### Actual result
The call is not routed.

### Nextcloud log

<summary>Nextcloud log</summary>

```
{"reqId":"3kcp14nywN3ontXVHlJN","level":2,"time":"2020-04-03T14:28:05+02:00","remoteAddr":"IP_ADDR","user":"IMPERSONATED_USER","app":"passwords","method":"GET","url":"/call/eipzbtrd","message":"Login attempt with invalid session for IMPERSONATED_USER","userAgent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:74.0) Gecko/20100101 Firefox/74.0","version":"18.0.3.0","id":"5e87305f359a0"}

{"reqId":"3kcp14nywN3ontXVHlJN","level":3,"time":"2020-04-03T14:28:05+02:00","remoteAddr":"IP_ADDR","user":"IMPERSONATED_USER","app":"index","method":"GET","url":"/call/eipzbtrd","message":{"Exception":"Exception","Message":"Unable to verify user IMPERSONATED_USER","Code":0,"Trace":[{"file":"/var/www/nextcloud/apps/passwords/lib/Services/EnvironmentService.php","line":300,"function":"loadUserInformation","class":"OCA\\Passwords\\Services\\EnvironmentService","type":"->","args":["IMPERSONATED_USER",{"__class__":"OC\\AppFramework\\Http\\Request"}]},{"file":"/var/www/nextcloud/apps/passwords/lib/Services/EnvironmentService.php","line":160,"function":"determineAppMode","class":"OCA\\Passwords\\Services\\EnvironmentService","type":"->","args":["IMPERSONATED_USER",{"__class__":"OC\\AppFramework\\Http\\Request"}]},{"function":"__construct","class":"OCA\\Passwords\\Services\\EnvironmentService","type":"->","args":["IMPERSONATED_USER",{"__class__":"OC\\AllConfig"},{"__class__":"OC\\AppFramework\\Http\\Request"},{"__class__":"OC\\Session\\CryptoSessionData"},{"__class__":"OCA\\Passwords\\Services\\LoggingService"},{"__class__":"OC\\Authentication\\Token\\Manager"},{"__class__":"OC\\User\\Manager"}]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":83,"function":"newInstanceArgs","class":"ReflectionClass","type":"->","args":[["IMPERSONATED_USER",{"__class__":"OC\\AllConfig"},{"__class__":"OC\\AppFramework\\Http\\Request"},{"__class__":"OC\\Session\\CryptoSessionData"},{"__class__":"OCA\\Passwords\\Services\\LoggingService"},{"__class__":"OC\\Authentication\\Token\\Manager"},{"__class__":"OC\\User\\Manager"}]]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":100,"function":"buildClass","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":[{"name":"OCA\\Passwords\\Services\\EnvironmentService","__class__":"ReflectionClass"}]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":115,"function":"resolve","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Passwords\\Services\\EnvironmentService"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php","line":415,"function":"query","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Passwords\\Services\\EnvironmentService"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php","line":386,"function":"queryNoFallback","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OCA\\Passwords\\Services\\EnvironmentService"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":70,"function":"query","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OCA\\Passwords\\Services\\EnvironmentService",true]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":100,"function":"buildClass","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":[{"name":"OCA\\Passwords\\Services\\ConfigurationService","__class__":"ReflectionClass"}]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":115,"function":"resolve","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Passwords\\Services\\ConfigurationService"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php","line":415,"function":"query","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Passwords\\Services\\ConfigurationService"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php","line":386,"function":"queryNoFallback","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OCA\\Passwords\\Services\\ConfigurationService"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":70,"function":"query","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OCA\\Passwords\\Services\\ConfigurationService",true]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":100,"function":"buildClass","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":[{"name":"OCA\\Passwords\\Helper\\Settings\\UserSettingsHelper","__class__":"ReflectionClass"}]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":115,"function":"resolve","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Passwords\\Helper\\Settings\\UserSettingsHelper"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php","line":415,"function":"query","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Passwords\\Helper\\Settings\\UserSettingsHelper"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php","line":386,"function":"queryNoFallback","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OCA\\Passwords\\Helper\\Settings\\UserSettingsHelper"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":70,"function":"query","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OCA\\Passwords\\Helper\\Settings\\UserSettingsHelper",true]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":100,"function":"buildClass","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":[{"name":"OCA\\Passwords\\Services\\UserSettingsService","__class__":"ReflectionClass"}]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":115,"function":"resolve","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Passwords\\Services\\UserSettingsService"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php","line":415,"function":"query","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Passwords\\Services\\UserSettingsService"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php","line":386,"function":"queryNoFallback","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OCA\\Passwords\\Services\\UserSettingsService"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":70,"function":"query","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OCA\\Passwords\\Services\\UserSettingsService",true]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":100,"function":"buildClass","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":[{"name":"OCA\\Passwords\\Services\\NotificationService","__class__":"ReflectionClass"}]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php","line":115,"function":"resolve","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Passwords\\Services\\NotificationService"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/DependencyInjection/DIContainer.php","line":415,"function":"query","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OCA\\Passwords\\Services\\NotificationService"]},{"file":"/var/www/nextcloud/lib/private/ServerContainer.php","line":134,"function":"queryNoFallback","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OCA\\Passwords\\Services\\NotificationService"]},{"file":"/var/www/nextcloud/lib/private/Notification/Manager.php","line":147,"function":"query","class":"OC\\ServerContainer","type":"->","args":["OCA\\Passwords\\Services\\NotificationService"]},{"file":"/var/www/nextcloud/lib/private/Notification/Manager.php","line":303,"function":"getNotifiers","class":"OC\\Notification\\Manager","type":"->","args":[]},{"file":"/var/www/nextcloud/apps/notifications/lib/Handler.php","line":148,"function":"dismissNotification","class":"OC\\Notification\\Manager","type":"->","args":[{"__class__":"OC\\Notification\\Notification"}]},{"file":"/var/www/nextcloud/apps/notifications/lib/Handler.php","line":111,"function":"deleteById","class":"OCA\\Notifications\\Handler","type":"->","args":[707,"IMPERSONATED_USER",{"__class__":"OC\\Notification\\Notification"}]},{"file":"/var/www/nextcloud/apps/notifications/lib/App.php","line":70,"function":"delete","class":"OCA\\Notifications\\Handler","type":"->","args":[{"__class__":"OC\\Notification\\Notification"}]},{"file":"/var/www/nextcloud/lib/private/Notification/Manager.php","line":283,"function":"markProcessed","class":"OCA\\Notifications\\App","type":"->","args":[{"__class__":"OC\\Notification\\Notification"}]},{"file":"/var/www/nextcloud/apps/spreed/lib/Controller/PageController.php","line":174,"function":"markProcessed","class":"OC\\Notification\\Manager","type":"->","args":[{"__class__":"OC\\Notification\\Notification"}]},{"file":"/var/www/nextcloud/apps/spreed/lib/Controller/PageController.php","line":118,"function":"index","class":"OCA\\Talk\\Controller\\PageController","type":"->","args":["eipzbtrd"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":170,"function":"showCall","class":"OCA\\Talk\\Controller\\PageController","type":"->","args":["eipzbtrd"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":99,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Talk\\Controller\\PageController"},"showCall"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/App.php","line":125,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Talk\\Controller\\PageController"},"showCall"]},{"file":"/var/www/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php","line":47,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Talk\\Controller\\PageController","showCall",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"token":"eipzbtrd","_route":"spreed.pagecontroller.showCall"}]},{"function":"__invoke","class":"OC\\AppFramework\\Routing\\RouteActionHandler","type":"->","args":[{"token":"eipzbtrd","_route":"spreed.pagecontroller.showCall"}]},{"file":"/var/www/nextcloud/lib/private/Route/Router.php","line":299,"function":"call_user_func","args":[{"__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"},{"token":"eipzbtrd","_route":"spreed.pagecontroller.showCall"}]},{"file":"/var/www/nextcloud/lib/base.php","line":1008,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/call/eipzbtrd"]},{"file":"/var/www/nextcloud/index.php","line":38,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/var/www/nextcloud/apps/passwords/lib/Services/EnvironmentService.php","Line":338,"CustomMessage":"--"},"userAgent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:74.0) Gecko/20100101 Firefox/74.0","version":"18.0.3.0","id":"5e87305f3598a"}
```

**Comment by marius-wieschollek** [comment]
You can try the nightly version of 2020.4.1. It should contain a fix for this.

**Comment by marius-wieschollek** [comment]
I have released 2020.4.1 and hope this fixes the issue.

**Comment by simonspa** [comment]
I can confirm that this issue is solved - thank you very much!
